crev-data/src/proof/content.rs

```diff
@@ -0,0 +1,128 @@
+use crate::Url;
+use chrono::{self, prelude::*};
+use failure::bail;
+
+use self::super::review;
+use self::super::trust::TrustDraft;
+use crate::proof;
+use crate::proof::{Proof, ProofType};
+use crate::Result;
+
+pub type Date = chrono::DateTime<FixedOffset>;
+
+pub trait ContentCommon: std::fmt::Debug + std::string::ToString + ContentCommonClone {
+    fn as_any(&self) -> &dyn std::any::Any;
+
+    fn date(&self) -> &Date;
+    fn set_date(&mut self, date: &Date);
+    fn date_utc(&self) -> chrono::DateTime<Utc> {
+        self.date().with_timezone(&Utc)
+    }
+
+    fn author(&self) -> &crate::PubId;
+    fn set_author(&mut self, id: &crate::PubId);
+
+    fn author_id(&self) -> crate::Id {
+        self.author().id.clone()
+    }
+
+    fn author_url(&self) -> Url {
+        self.author().url.clone()
+    }
+
+    fn draft_title(&self) -> String;
+    fn validate_data(&self) -> Result<()>;
+
+    fn parse(s: &str) -> Result<Self>
+    where
+        Self: Sized;
+
+    fn parse_draft(&self, s: &str) -> Result<Self>
+    where
+        Self: Sized;
+
+    fn proof_type(&self) -> ProofType;
+    fn to_draft_string(&self) -> String;
+
+    fn sign_by(&self, id: &crate::id::OwnId) -> Result<Proof> {
+        let body = self.to_string();
+        let signature = id.sign(&body.as_bytes());
+        Ok(Proof {
+            digest: crev_common::blake2b256sum(&body.as_bytes()),
+            body,
+            signature: crev_common::base64_encode(&signature),
+            content: self.clone_box(),
+        })
+    }
+
+    fn ensure_serializes_to_valid_proof(&self) -> Result<()> {
+        let body = self.to_string();
+        let signature = "somefakesignature";
+        let proof = proof::Proof {
+            digest: crev_common::blake2b256sum(&body.as_bytes()),
+            body,
+            signature: crev_common::base64_encode(&signature),
+            content: self.clone_box(),
+        };
+        let parsed = proof::Proof::parse(std::io::Cursor::new(proof.to_string().as_bytes()))?;
+
+        if parsed.len() != 1 {
+            bail!("Serialized to {} proofs", parsed.len());
+        }
+
+        Ok(())
+    }
+}
+
+pub trait ContentCommonClone {
+    fn clone_box(&self) -> Box<dyn ContentCommon>;
+}
+
+impl<T: 'static + ContentCommon + std::clone::Clone> ContentCommonClone for T {
+    fn clone_box(&self) -> Box<dyn ContentCommon> {
+        Box::new(self.clone())
+    }
+}
+
+impl std::clone::Clone for Box<dyn ContentCommon> {
+    fn clone(&self) -> Box<dyn ContentCommon> {
+        self.clone_box()
+    }
+}
+
+pub fn parse_draft(content: Box<dyn ContentCommon>, s: &str) -> Result<Box<dyn ContentCommon>> {
```

Comment:
This should probably follow a similiar pattern as `ProofParser`, or ContentCommon` should have a `get_draft_parser(&self) -> Box<dyn DraftParser>` function.